model_analyzer/analyzer.py

```diff
@@ -0,0 +1,227 @@
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import time
+
+from model_analyzer.model_analyzer_exceptions import TritonModelAnalyzerException
+from .monitor.dcgm.dcgm_monitor import DCGMMonitor
+from .perf_analyzer.perf_analyzer import PerfAnalyzer
+from .perf_analyzer.perf_config import PerfAnalyzerConfig
+from .record.record_aggregator import RecordAggregator
+from .record.gpu_free_memory import GPUFreeMemory
+from .record.gpu_used_memory import GPUUsedMemory
+from .record.perf_throughput import PerfThroughput
+from .record.perf_latency import PerfLatency
+from .output.output_table import OutputTable
+
+
+class Analyzer:
+    """
+    A class responsible for coordinating
+    the various components of the model_analyzer.
+    Configured with metrics to monitor,
+    exposes profiling and result writing
+    methods.
+    """
+
+    def __init__(self, args, monitoring_metrics, writer):
+        """
+        Parameters
+        ----------
+        args : namespace
+            The arguments passed into the CLI
+        monitoring_metrics : List of Record types
+            The list of metric types to monitor.
+        writer : OutputWriter
+            Used to write the result tables to an output stream
+        """
+
+        self.perf_analyzer_path = args.perf_analyzer_path
+        self.monitoring_interval = args.monitoring_interval
+        self.monitoring_metrics = monitoring_metrics
+        self.writer = writer
+        self.param_headers = ['Model', 'Batch', 'Concurrency']
+
+        # Separates metric tags into perf_analyzer related and DCGM related tags
+        self.dcgm_tags = []
+        self.perf_tags = []
+        for metric in self.monitoring_metrics:
+            if metric in [GPUFreeMemory, GPUUsedMemory]:
```

Comment:
I think it would be great if we use `list(DCGMMonitor.model_analyzer_to_dcgm_field)` so that we wouldn't have to update this list every time we add a new metric.